server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/AllocationBalancingRoundMetrics.java

```diff
@@ -0,0 +1,113 @@
+    public static final String NUMBER_OF_SHARD_MOVES_HISTOGRAM_METRIC_NAME = "es.allocator.balancing_round.shard_moves.histogram";
+
+    // histograms that measure current utilization
+    public static final String NUMBER_OF_SHARDS_METRIC_NAME = "es.allocator.balancing_round.shard_count.histogram";
+    public static final String DISK_USAGE_BYTES_METRIC_NAME = "es.allocator.balancing_round.disk_usage_bytes.histogram";
+    public static final String WRITE_LOAD_METRIC_NAME = "es.allocator.balancing_round.write_load.histogram";
+    public static final String TOTAL_WEIGHT_METRIC_NAME = "es.allocator.balancing_round.total_weight.histogram";
+
+    private final LongCounter balancingRoundCounter;
+    private final LongCounter shardMovesCounter;
+    private final LongHistogram shardMovesHistogram;
+
+    private final LongHistogram shardCountHistogram;
+    private final DoubleHistogram diskUsageHistogram;
+    private final DoubleHistogram writeLoadHistogram;
+    private final DoubleHistogram totalWeightHistogram;
+
+    public static AllocationBalancingRoundMetrics NOOP = new AllocationBalancingRoundMetrics(MeterRegistry.NOOP);
+
+    public AllocationBalancingRoundMetrics(MeterRegistry meterRegistry) {
+        this.balancingRoundCounter = meterRegistry.registerLongCounter(
+            NUMBER_OF_BALANCING_ROUNDS_METRIC_NAME,
+            "Total number of balancing rounds",
+            "unit"
+        );
+        this.shardMovesCounter = meterRegistry.registerLongCounter(
+            NUMBER_OF_SHARD_MOVES_METRIC_NAME,
+            "Total number of shard moves",
+            "unit"
+        );
+
+        this.shardMovesHistogram = meterRegistry.registerLongHistogram(
+            NUMBER_OF_SHARD_MOVES_HISTOGRAM_METRIC_NAME,
+            "Number of shard movements executed in a balancing round",
+            "unit"
+        );
+        this.shardCountHistogram = meterRegistry.registerLongHistogram(
+            NUMBER_OF_SHARDS_METRIC_NAME,
+            "change in node shard count per balancing round",
+            "unit"
+        );
+        this.diskUsageHistogram = meterRegistry.registerDoubleHistogram(
+            DISK_USAGE_BYTES_METRIC_NAME,
+            "change in disk usage in bytes per balancing round",
+            "unit"
+        );
+        this.writeLoadHistogram = meterRegistry.registerDoubleHistogram(
+            WRITE_LOAD_METRIC_NAME,
+            "change in write load per balancing round",
+            "1.0"
+        );
+        this.totalWeightHistogram = meterRegistry.registerDoubleHistogram(
+            TOTAL_WEIGHT_METRIC_NAME,
+            "change in total weight per balancing round",
+            "1.0"
+        );
+    }
+
+    @SuppressForbidden(reason = "ForbiddenAPIs bans Math.abs on longs, but long.MIN_VALUE is impossible here")
+    private long longAbsNegativeSafe(long input) {
+        return Math.abs(input);
```

Comment:
Can we just put an assertion above the call to abs that ensures input isn't `Long.MIN_VALUE`